dbms/src/Storages/StorageMySQL.cpp

```diff
@@ -0,0 +1,187 @@
+#include <Storages/StorageMySQL.h>
+
+#include <Interpreters/Context.h>
+#include <Interpreters/ExpressionAnalyzer.h>
+
+#include <Dictionaries/MySQLBlockInputStream.h>
+
+#include <Parsers/ASTSelectQuery.h>
+
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+
+namespace DB
+{
+
+/** Implements storage in the MySQL database.
+  * Use ENGINE = mysql(host_port, database_name, table_name, user_name, password)
+  * Read only
+  */
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_FIND_FIELD;
+};
+
+StorageMySQL::StorageMySQL(
+    const std::string & table_name_,
+    const std::string & server_,
+    int port_,
+    const std::string & database_name_,
+    const std::string & mysql_table_name_,
+    const std::string & user_name_,
+    const std::string & password_,
+    const NamesAndTypesListPtr & columns_,
+    const NamesAndTypesList & materialized_columns_,
+    const NamesAndTypesList & alias_columns_,
+    const ColumnDefaults & column_defaults_,
+    const Context & context_)
+    : IStorage(materialized_columns_, alias_columns_, column_defaults_),
+    table_name(table_name_), server(server_), port(port_), database_name(database_name_), mysql_table_name(mysql_table_name_), user_name(user_name_), password(password_), columns(columns_), context_global(context_),
+    pool(database_name_, server_, user_name_, password_, port_)
+{
+    column_map.set_empty_key("");
+    for (const auto & it: *columns)
+    {
+        column_map[it.name] = it.type;
+    }
+}
+
+bool isCompatible(const IAST & where)
+{
+    String name = where.getID();
+    if ((name == "Function_and") || (name == "Function_or"))
```

Comment:
It's better to cast `IAST` to `ASTFunction` and compare by `ASTFunction::name` member. `typeid_cast` is cheaper than string comparison and `getID` function has other semantic.

Reply:
Fixed